src/main/java/io/apicurio/datamodels/asyncapi/models/AaiSchema.java

```diff
@@ -130,50 +133,103 @@ public void setExternalDocumentation(ExternalDocumentation externalDocs) {
     }
 
     /**
-     * Gets a list of all the properties.
+     * @see IPropertyParent#getProperties() 
      */
-    public List<AaiSchema> getProperties() {
-        List<AaiSchema> rval = new ArrayList<>();
+    @Override
+    public List<Schema> getProperties() {
+        List<Schema> rval = new ArrayList<>();
         if (this.properties != null) {
             rval.addAll(this.properties.values());
         }
         return rval;
     }
 
     /**
-     * Add a property.
-     * @param propertyName
-     * @param schema
+     * @see IPropertyParent#addProperty(String, Schema) 
      */
-    public AaiSchema addProperty(String propertyName, AaiSchema schema) {
+    @Override
+    public Schema addProperty(String propertyName, Schema schema) {
         if (this.properties == null) {
             this.properties = new LinkedHashMap<>();
         }
-        this.properties.put(propertyName, schema);
+        this.properties.put(propertyName, ((AaiSchema) schema));
         return schema;
     }
 
     /**
-     * Removes a property by name.
-     * @param propertyName
+     * @see IPropertyParent#removeProperty(String) 
      */
-    public AaiSchema removeProperty(String propertyName) {
+    @Override
+    public Schema removeProperty(String propertyName) {
         if (this.properties != null) {
             return this.properties.remove(propertyName);
         }
         return null;
     }
 
     /**
-     * Gets a single property.
-     * @param propertyName
+     * @see IPropertyParent#getProperty(String) 
      */
-    public AaiSchema getProperty(String propertyName) {
+    @Override
+    public Schema getProperty(String propertyName) {
         if (this.properties != null) {
             return this.properties.get(propertyName);
         }
         return null;
     }
+
+    /**
+     * @see IPropertyParent#getRequiredProperties()
+     */
+    @Override
+    public List<String> getRequiredProperties() {
+        return this.required;
+    }
+
+    /**
+     * @see IPropertyParent#setRequiredProperties(List) 
+     */
+    @Override
+    public void setRequiredProperties(List<String> requiredProperties) {
+        this.required = requiredProperties;
+    }
+
+    /**
+     * @see IPropertyParent#isPropertyRequired(String)
+     */
+    @Override
+    public boolean isPropertyRequired(String propertyName) {
+        if (this.required != null) {
```

Comment:
This should work but it's probably better to use ModelUtils.isDefined

Reply:
Same with other null checks in this file.

Reply:
Same for OasSchema as well